crates/runtime/src/instance/allocator.rs

```diff
@@ -32,28 +32,16 @@ pub use self::pooling::{
 
 /// Represents a request for a new runtime instance.
 pub struct InstanceAllocationRequest<'a> {
-    /// The module being instantiated.
-    pub module: &'a Arc<Module>,
-
-    /// The unique ID of the module being allocated within this engine.
-    pub unique_id: Option<CompiledModuleId>,
-
-    /// The base address of where JIT functions are located.
-    pub image_base: usize,
-
-    /// If using MemFD-based memories, the backing MemFDs.
-    pub memfds: Option<&'a Arc<ModuleMemFds>>,
-
-    /// Descriptors about each compiled function, such as the offset from
-    /// `image_base`.
-    pub functions: &'a PrimaryMap<DefinedFuncIndex, FunctionInfo>,
+    /// The info related to the compiled version of this module,
+    /// needed for instantiation: function metadata, JIT code
+    /// addresses, precomputed images for lazy memory and table
+    /// initialization, and the like. This Arc is cloned and held for
+    /// the lifetime of the instance.
+    pub runtime_info: &'a Arc<dyn ModuleRuntimeInfo + 'static>,
```

Comment:
FWIW the `+ 'static` can be removed here (and in other locations probably) since if you don't mention anything that's the default.

Reply:
Done!